nginx/nginx.go

```diff
@@ -443,6 +443,15 @@ func upstreamID(e controller.IngressEntry) string {
 	return fmt.Sprintf("%s.%s.%d", e.Namespace, e.ServiceAddress, e.ServicePort)
 }
 
+func (s server) HasRootLocation() bool {
+	for i := range s.Locations {
+		if s.Locations[i].Path == "/" {
+			return true
+		}
+	}
+	return false
```

Comment:
@totahuanocotl I was thinking about but in this case, the cost is not so big because we don't create copies (iterating over the array index). 

Doing this iteration during the creation of the template: 
- We don't change the original entries which are taken from the external sources. 
- It is always one-to-one relationship between entries for template and the original data.
- It is clear from reading the template and the name of this function why this entry appeared in the result.  

In my opinion, adding the additional entry to the server entries would make the solution harder to read and change in the future.

Reply:
hey @robertfirek, maybe I wasn't clear on my comment, as I didn't mean adding or changing location entries in the code. 
I just meant figuring out if there is a root path present or not when you create the server entries, and keep it as a boolean property, and then just use it as you have in the template.
While you process the ingresses in the [createServerEntries](
https://github.com/sky-uk/feed/blob/f21a3ec0fe33b4ab54b018e045c70ff643ed6012/nginx/nginx.go#L489), if the location is "/", you can just set the server's property `hasRootPath` to true.

Reply:
What is the advantage of your solution? 
My reasoning was to avoid adding additional field because the value would be very easily calculated during the template creation (no modifications in the original set) and it is resolved only when it is needed. 

I think we adding the logic to the iteration would require more complex code because there is no guarantee that all entries for the same server entry will come each after another and we would need to introduce additional map with information about root path per server.

Reply:
I was just trying to avoid having to iterate over every path of every server to figure out if there is a root path defined when you have already iterated through all of them once already. I know as it currently

You don't need extra logic, you only need to look at the current path, because there is already a map that handles the location-server info. If the current path for a server is root, then the server has a root path. If by the time you finished processing all the entries there was no root path for a server, then the property would not have been set to true. 

Just want to point out that it is a suggestion, not  a required change.